src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -225,316 +227,332 @@ where
+                    use futures::future::Either;
+                    let update = match changes {
+                        Either::Left((update, _)) => update,
+                        Either::Right((Some(Event::Progress(resume_frontier_update)), _)) => {
+                            // The first message from the resumption frontier source
+                            // could be the same frontier as the initialization frontier, so we
+                            // just move on
+                            if PartialOrder::less_equal(
+                                &resume_frontier_update,
+                                &initial_resume_upper,
+                            ) {
+                                continue;
+                            }
+                            tracing::trace!(
+                                %id,
+                                resumption_frontier = ?resume_frontier_update,
+                                "received new resumption frontier"
+                            );
+                            continue;
+                        }
+                        _ => {
+                            continue;
+                        }
+                    };
 
-            while let Poll::Ready(res) = source_reader.as_mut().poll_next(&mut context) {
-                let update = match res {
-                    Some(update) => update,
-                    None => {
-                        return;
-                    }
-                };
-                let (messages, non_definite_errors, unconsumed_partitions, source_upper) =
-                    match update {
+                    let update = match update {
                         Some(update) => update,
                         None => {
-                            trace!("source_reader({id}) {worker_id}/{worker_count}: is terminated");
-                            // We will never produce more data, clear our capabilities to
-                            // communicate this downstream.
-                            cap_set.downgrade(&[]);
                             return;
                         }
                     };
+                    let (messages, non_definite_errors, unconsumed_partitions, source_upper) =
+                        match update {
+                            Some(update) => update,
+                            None => {
+                                trace!(
+                                    "source_reader({id}) {worker_id}/{worker_count}: is terminated"
+                                );
+                                // We will never produce more data, clear our capabilities to
+                                // communicate this downstream.
+                                cap_set.downgrade(&[]);
+                                return;
+                            }
+                        };
 
-                trace!(
+                    trace!(
```

Comment:
Something is weird with the indentation of this trace

Reply:
ugh, rustfmt is so bad at this